spacy/tests/test_requirements.py

```diff
@@ -0,0 +1,59 @@
+import re
+from pathlib import Path
+
+
+def test_build_dependencies(en_vocab):
+    # Check that library requirements are pinned exactly the same across different setup files.
+    libs_ignore_requirements = ["pytest", "pytest-timeout", "mock", "flake8", "jsonschema"]
+    libs_ignore_setup = ["fugashi", "natto-py", "pythainlp"]
+
+    # check requirements.txt
+    root_dir = Path(__file__).parent.parent.parent
+    req_file = root_dir / "requirements.txt"
+    req_dict = {}
+    with req_file.open() as f:
+        lines = f.readlines()
+    for line in lines:
+        line = line.strip()
+        if not line.startswith("#"):
+            lib, v = _parse_req(line)
+            if lib and lib not in libs_ignore_requirements:
+                req_dict[lib] = v
+
+    # check setup.cfg and compare to requirements.txt
+    # also fails when there are missing or additional libs
+    setup_file = root_dir / "setup.cfg"
```

Comment:
Not sure how much this helps, but Python's [configparser](https://docs.python.org/3/library/configparser.html) should be able to read the `.cfg` files. I think this might let us drop some of the hack code here?

Reply:
You mean you're not a huge fan of the hacky regular expressions? :-)

Reply:
Well, not if they're already built into the standard library 😛

Reply:
I will clean this code up once we figure out the trouble with the CI ;-)